crates/llvm-context/src/polkavm/mod.rs

```diff
@@ -22,64 +25,51 @@ pub fn initialize_target() {
     }
 
-    /*
-    let mut assembly =
-        zkevm_assembly::Assembly::from_string(assembly_text.to_owned(), metadata_hash).map_err(
-            |error| {
-                anyhow::anyhow!(
-                    "The contract `{}` assembly parsing error: {}",
-                    contract_path,
-                    error,
-                )
-            },
-        )?;
-
-    let bytecode_words = match zkevm_assembly::get_encoding_mode() {
-        zkevm_assembly::RunningVmEncodingMode::Production => { assembly.compile_to_bytecode_for_mode::<8, zkevm_opcode_defs::decoding::EncodingModeProduction>() },
-        zkevm_assembly::RunningVmEncodingMode::Testing => { assembly.compile_to_bytecode_for_mode::<16, zkevm_opcode_defs::decoding::EncodingModeTesting>() },
-    }
+    let bytecode = hex::decode(encoded_hex_text)
+        .map_err(|e| anyhow::anyhow!("Failed to decode encoded hex text:\n{}\n", e))?;
+
+    let program_blob = ProgramBlob::parse(bytecode.as_slice())
+        .map_err(|error| anyhow::anyhow!(format!("Failed to parse program blob:\n{}\n", error)))?;
+
+    let mut disassembler =
+        Disassembler::new(&program_blob, DisassemblyFormat::Guest).map_err(|error| {
+            anyhow::anyhow!(format!(
+                "Failed to create disassembler for contract:\n{:?}\n\nDue to:\n{}\n",
+                contract_path, error
+            ))
+        })?;
+
+    let mut disassembled_code = Vec::new();
+    disassembler
+        .disassemble_into(&mut disassembled_code)
         .map_err(|error| {
-            anyhow::anyhow!(
-                "The contract `{}` assembly-to-bytecode conversion error: {}",
+            anyhow::anyhow!(format!(
+                "Failed to disassemble contract:\n{:?}\n\nDue to:\n{}\n\nGas details:{:?}\n",
                 contract_path,
                 error,
-            )
+                disassembler.display_gas()
+            ))
         })?;
 
-    let bytecode_hash = match zkevm_assembly::get_encoding_mode() {
-        zkevm_assembly::RunningVmEncodingMode::Production => {
-            zkevm_opcode_defs::utils::bytecode_to_code_hash_for_mode::<
-                8,
-                zkevm_opcode_defs::decoding::EncodingModeProduction,
-            >(bytecode_words.as_slice())
-        }
-        zkevm_assembly::RunningVmEncodingMode::Testing => {
-            zkevm_opcode_defs::utils::bytecode_to_code_hash_for_mode::<
-                16,
-                zkevm_opcode_defs::decoding::EncodingModeTesting,
-            >(bytecode_words.as_slice())
-        }
-    }
-    .map(hex::encode)
-    .map_err(|_error| {
-        anyhow::anyhow!("The contract `{}` bytecode hashing error", contract_path,)
+    let assembly_text = String::from_utf8(disassembled_code).map_err(|error| {
+        anyhow::anyhow!(format!(
+            "Failed to convert disassembled code to string for contract\n{:?}\n\nDue to:\n{}\n",
+            contract_path, error
+        ))
```

Comment:
```suggestion
    let assembly_text = String::from_utf8(disassembled_code)
        .context("Failed to convert disassembled code to string for contract: {contract_path}")?;
```

Same nitpick here 

Note that for this to work you'll have to import the trait like `use anyhow::Context as AnyhowContext` or something like this, as we already import the `Context` from this crate (aliasing the anyhow context like this neatly resolves the conflict).